openmc/tallies.py

```diff
@@ -1530,43 +1585,28 @@ def _outer_product(self, other, binary_op):
         if self_copy.num_realizations == other_copy.num_realizations:
             new_tally.num_realizations = self_copy.num_realizations
 
-        # If filters are identical, simply reuse them in derived tally
-        if self_copy.filters == other_copy.filters:
+        # Add filters to the new tally
+        if filter_product == 'entrywise':
             for self_filter in self_copy.filters:
                 new_tally.add_filter(self_filter)
-
-        # Generate filter "outer products" for non-identical filters
         else:
+            all_filters = [self_copy.filters, other_copy.filters]
+            for self_filter, other_filter in itertools.product(*all_filters):
+                new_filter = CrossFilter(self_filter, other_filter, binary_op)
+                new_tally.add_filter(new_filter)
 
-            # Find the common longest sequence of shared filters
-            match = 0
-            for self_filter, other_filter in zip(self_copy.filters, other_copy.filters):
-                if self_filter == other_filter:
-                    match += 1
-                else:
-                    break
-
-            match_filters = self_copy.filters[:match]
-            cross_filters = [self_copy.filters[match:], other_copy.filters[match:]]
-
-            # Simply reuse shared filters in derived tally
-            for filter in match_filters:
-                new_tally.add_filter(filter)
-
-            # Use cross filters to combine non-shared filters in derived tally
-            if len(self_copy.filters) != match and len(other_copy.filters) == match:
-                for filter in cross_filters[0]:
-                    new_tally.add_filter(filter)
-            elif len(self_copy.filters) == match and len(other_copy.filters) != match:
-                for filter in cross_filters[1]:
-                    new_tally.add_filter(filter)
-            else:
-                for self_filter, other_filter in itertools.product(*cross_filters):
-                    new_filter = CrossFilter(self_filter, other_filter, binary_op)
-                    new_tally.add_filter(new_filter)
+        # Add nuclides to the new tally
+        if nuclide_product == 'entrywise':
+            for self_nuclide in self_copy.nuclides:
+                new_tally.add_nuclide(self_nuclide)
+        else:
+            all_nuclides = [self_copy.nuclides, other_copy.nuclides]
+            for self_nuclide, other_nuclide in itertools.product(*all_nuclides):
+                new_nuclide = CrossNuclide(self_nuclide, other_nuclide, binary_op)
+                new_tally.add_nuclide(new_nuclide)
 
-        # Generate score "outer products"
-        if self_copy.scores == other_copy.scores:
+        # Add scores to the new tally
+        if score_product == 'entrywise':
             new_tally.num_score_bins = self_copy.num_score_bins
             for self_score in self_copy.scores:
                 new_tally.add_score(self_score)
```

Comment:
In `input_xml.F90`,  a score of `n2n` is converted to the reaction type 16 so I don't really see a way (or need) to differentiate n2n from 16 in the statepoint/summary files. However, this does complicate finding a fix for the duplicate scores issues since the scores need to first be converted to reaction types and then the collection of reaction types needs to be analyzed and duplicates removed.